### PR TITLE
Add stripe link headers to requests

### LINKED ISSUE #2366: StripeLink: Add beta headers for Stripe requests
If StripeLink payment method is enabled, we should add `link_beta=v1` to the `Stripe-Version` header request on Stripe, when the payment intent is created and on confirmPayment.
So, the request header will be: `Stripe-Version: 2020-08-27;link_beta=v1`

### PR DESCRIPTION
**Pull request by dpaun1985**
Fixes #2366 

## Changes proposed in this Pull Request:
When Stripe Link is enabled, the requests to Stripe should have `link_autofill_modal_beta_1` and `link_beta_2` headers. Also api version should be `2020-08-27;link_beta=v1`.

## Testing instructions
- all tests should pass. 
- create orders with and without UPE enabled. We should not encounter any errors.

---

-   [ ] Covered with tests (or have a good reason not to test in description ☝️)
-   [ ] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [ ] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)

### CODE REVIEW

File: client/api/index.js [hunk 1]
```diff
@@ -58,23 +58,46 @@ export default class WCStripeAPI {
 	 * @return {Object} The Stripe Object.
 	 */
 	getStripe() {
-		const { key, locale, isUPEEnabled } = this.options;
+		const { key, locale, isUPEEnabled, paymentMethodsConfig } = this.options;
```
Comment (ricardo) [inline client/api/index.js]: You probably already noticed, but the linter suggests breaking this into multiple lines.

File: client/api/index.js [hunk 2]
```diff
@@ -58,23 +58,46 @@ export default class WCStripeAPI {
...
+		const isStripeLinkEnabled =
+			undefined !== paymentMethodsConfig.card &&
+			undefined !== paymentMethodsConfig.link;
```
Comment (ricardo) [inline client/api/index.js:69]: Does `card` need to be enabled for `link` to be enabled, or can it work separately, like other UPE payment methods?

File: client/api/index.js [hunk 3]
```diff
@@ -58,23 +58,46 @@ export default class WCStripeAPI {
...
 		if ( ! this.stripe ) {
 			if ( isUPEEnabled ) {
-				this.stripe = new Stripe( key, {
-					betas: [ 'payment_element_beta_1' ],
-					locale,
-				} );
+				let betas = [];
+				if ( isUPEEnabled ) {
+					betas.push( 'payment_element_beta_1' );
+				}
```
Comment (ricardo) [inline client/api/index.js]: We already have a check `if ( isUPEEnabled )` right above, so no need for this second `if`.

We can do `let betas = [ 'payment_element_beta_1' ];`.

File: client/api/index.js [hunk 4]
```diff
@@ -58,23 +58,46 @@ export default class WCStripeAPI {
...
 		}
 		return this.stripe;
 	}
 
+	createStripe( key, locale, betas = [] ) {
+		const options = { locale };
+
+		if ( betas ) {
+			options.betas = betas;
+		}
```
Comment (ricardo) [inline client/api/index.js]: This if is redundant as `if ( [] )` should validate as `true`, so we can either do `if ( betas.length )` or if it doesn't matter to have an empty array, `const options = { locale, betas };`.

File: client/api/index.js [hunk 5]
```diff
@@ -58,23 +58,46 @@ export default class WCStripeAPI {
...
+
+		if ( betas.includes( 'link_beta_2' ) ) {
+			options.apiVersion = '2020-08-27;link_beta=v1';
```
Comment (ricardo) [inline client/api/index.js:97]: Just as a side note, no need to change anything now, but maybe we will need to change the [back-end Stripe API version](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/db315182e581a89b5a177c4d485c39f6cf8b77f6/includes/class-wc-stripe-api.php#L17) as well? 🤔

**Review by ricardo — CHANGES_REQUESTED** [review]
Looking good.

UPE still works for me after the change.

File: client/api/index.js (hunk 2 above)
Comment (dpaun1985) [reply client/api/index.js:69]: `link` can work only with `card` payment method. So yes, we need both payment methods enabled.

File: client/api/index.js (hunk 5 above)
Comment (dpaun1985) [reply client/api/index.js:97]: I expect that we don't need the new api version in backend, but if we will need it, I will change it at that time.

**Comment by dpaun1985** [comment]
> Looking good.
> 
> UPE still works for me after the change.

Thank you @ricardo !
I made the required changes. Could you please check again the PR ?

**Review by ricardo — APPROVED** [review]
LGTM 👍

**Review by mordeth — APPROVED** [review]
LGTM! ✅

**Pull request merged**